makelib/golang.mk

```diff
@@ -30,10 +30,6 @@ GO_SUBDIRS ?= cmd pkg
 # Optional. Additional subdirs used for integration or e2e testings
 GO_INTEGRATION_TESTS_SUBDIRS ?=
 
-# Optional directories (relative to CURDIR)
-GO_VENDOR_DIR ?= vendor
-GO_PKG_DIR ?= $(WORK_DIR)/pkg
```

Comment:
Use the default/system global `$GOPATH/pkg` directory, this helps go mod cache (`$GOPATH/pkg/mod`) reuse between different repos.

Reply:
Not to mention having `$(WORK_DIR)/pkg` messes up with vscode + gopls when it sees several thousand files.

Reply:
I think I'm mostly concerned about this part of the changes. I agree with the cache reuse aspect; however I want to dig into _why_ the mod cache was placed in the directories to get that context.

Reply:
>however I want to dig into why the mod cache was placed in the directories to get that context

To tie a bow on this, it looks like this behavior was introduced during the introduction to using Go modules here https://github.com/upbound/build/pull/74. Most likely this was done for two reasons:
1. we were still supporting dep at that time
2. it wasn't entirely clear how to work with go modules _yet_

With that context, I'm comfortable moving forward with this changeset.